tests: fix failures on the CI

Tests are passing locally but not on the GitHub CI:

```
=========================== short test summary info ============================
FAILED tests/unit/test_app.py::test_static_image - AssertionError: assert False
 +  where False = <built-in method startswith of str object at 0x7fd1ee708030>('image/')
 +    where <built-in method startswith of str object at 0x7fd1ee708030> = ''.startswith
 +      where '' = Content-Length: 26714\nLast-Modified: Mon, 16 Sep 2024 21:54:47 GMT\nDate: Mon, 16 Sep 2024 21:54:47 GMT\nEtag: 16e4f0585ed4c334b75e29ea7c982b852a9d2c1d\nAccept-Ranges: bytes\nCache-Control: public, max-age=31536000\nContent-Type: text/html; charset=UTF-8\n.content_type
FAILED tests/unit/test_fn_sync.py::test_sync_feed[True] - AssertionError: assert 2 == 3
 +  where 2 = <function sync_feed at 0x7fd1d6b30e50>('https://shaarli.example.org/')
 +    where <function sync_feed at 0x7fd1d6b30e50> = helpers.sync_feed
FAILED tests/unit/test_fn_sync.py::test_sync_feed[False] - AssertionError: assert 2 == 3
 +  where 2 = <function sync_feed at 0x7fd1d6b30e50>('https://shaarli.example.org/')
 +    where <function sync_feed at 0x7fd1d6b30e50> = helpers.sync_feed
======================== 3 failed, 145 passed in 2.00s =========================
```

--- a/host/app.py
+++ b/host/app.py
@@ -178,7 +178,7 @@ def static_favicon() -> HTTPResponse:
 
 
 @route("/image/<image>")
-def static_image(image: str) -> str:
+def static_image(image: str) -> HTTPResponse:
     """Get an image."""
     response = static_file(image, root=constants.IMAGES)
     response.set_header("Cache-Control", "public, max-age=31536000")

--- a/tests/unit/conftest.py
+++ b/tests/unit/conftest.py
@@ -57,7 +57,7 @@ def setup_data(tmp_path: Path) -> None:
     # Create the JSON feed
     feed_key = functions.small_hash(functions.feed_key(FEED_URL))
     data = {}
-    for idx, (file, file_size, size, thumb_size, color, checksum) in enumerate(TEST_IMAGES, 1):
+    for idx, (file, file_size, size, thumb_size, color, checksum, mimetype) in enumerate(TEST_IMAGES, 1):
         url = f"{FEED_URL}/{file.name}"
         cache_key = functions.small_hash(url)
         stored_file = f"{cache_key}{file.suffix}"

--- a/tests/unit/constants.py
+++ b/tests/unit/constants.py
@@ -16,7 +16,7 @@ IMAGE_SQUARE = DATA / "3.jpg"
 IMAGE_WEBP = DATA / "4.webp"
 
 TEST_IMAGES = [
-    # (local file, file size, original size, thumbnail size, docolav, checksum)
+    # (local file, file size, original size, thumbnail size, docolav, checksum, mimetype)
     (
         IMAGE_JPG,
         22548,
@@ -24,6 +24,7 @@ TEST_IMAGES = [
         Size(width=400, height=267),
         "585E50",
         "29c25875e30fe9547349057887b19682",
+        "image/jpeg",
     ),
     (
         IMAGE_PNG,
@@ -32,6 +33,7 @@ TEST_IMAGES = [
         Size(width=317, height=400),
         "B19C95",
         "2a4ab87892014129a4f1e73ba9dfe22d",
+        "image/png",
     ),
     (
         IMAGE_SQUARE,
@@ -40,6 +42,7 @@ TEST_IMAGES = [
         Size(width=400, height=400),
         "656B72",
         "8bdecafe89fd9e02981720ba8172dad4",
+        "image/jpeg",
     ),
     (
         IMAGE_WEBP,
@@ -48,6 +51,7 @@ TEST_IMAGES = [
         Size(width=358, height=400),
         "6A7777",
         "95603fa0936de10e5068b1d200680273",
+        "image/webp",
     ),
 ]
 FEED_URL = "https://shaarli.example.org"

--- a/tests/unit/test_app.py
+++ b/tests/unit/test_app.py
@@ -269,8 +269,10 @@ def test_static_favicon() -> None:
 
 def test_static_image(setup_data: FixtureFunction) -> None:
     image = choice(functions.retrieve_all_uniq_metadata())
+    print(image)
     response = app.static_image(image.file)
     response.body.close()
+    print(response)
     assert response.status_code == 200
     assert response.content_type.startswith("image/")
 

--- a/tests/unit/test_app.py
+++ b/tests/unit/test_app.py
@@ -269,10 +269,8 @@ def test_static_favicon() -> None:
 
 def test_static_image(setup_data: FixtureFunction) -> None:
     image = choice(functions.retrieve_all_uniq_metadata())
-    print(image)
     response = app.static_image(image.file)
     response.body.close()
-    print(response)
     assert response.status_code == 200
     assert response.content_type.startswith("image/")
 

--- a/tests/unit/test_app.py
+++ b/tests/unit/test_app.py
@@ -271,6 +271,7 @@ def test_static_image(setup_data: FixtureFunction) -> None:
     image = choice(functions.retrieve_all_uniq_metadata())
     response = app.static_image(image.file)
     response.body.close()
+    print(response)
     assert response.status_code == 200
     assert response.content_type.startswith("image/")
 

--- a/tests/unit/test_app.py
+++ b/tests/unit/test_app.py
@@ -269,9 +269,9 @@ def test_static_favicon() -> None:
 
 def test_static_image(setup_data: FixtureFunction) -> None:
     image = choice(functions.retrieve_all_uniq_metadata())
+    assert image
     response = app.static_image(image.file)
     response.body.close()
-    print(response)
     assert response.status_code == 200
     assert response.content_type.startswith("image/")
 

--- a/tests/unit/test_fn_images.py
+++ b/tests/unit/test_fn_images.py
@@ -78,7 +78,7 @@ def test_docolav(
     thumb_size: Size,
     color: str,
     checksum: str,
-    mimitype: str,
+    mimetype: str,
 ) -> None:
     assert functions.docolav(file) == color
 
@@ -91,6 +91,6 @@ def test_get_size(
     thumb_size: Size,
     color: str,
     checksum: str,
-    mimitype: str,
+    mimetype: str,
 ) -> None:
     assert functions.get_size(file) == size

--- a/tests/unit/test_fn_images.py
+++ b/tests/unit/test_fn_images.py
@@ -33,7 +33,7 @@ def test_checksum(
 
 @pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
 def test_create_thumbnail(
-    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimitype: str, tmp_path: Path
+    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimetype: str, tmp_path: Path
 ) -> None:
     dest_file = tmp_path / file.name
     assert not dest_file.is_file()

--- a/tests/unit/test_fn_images.py
+++ b/tests/unit/test_fn_images.py
@@ -24,16 +24,16 @@ def test_any_css_class_question_gray_pride() -> None:
     assert functions.any_css_class_question() == "gay-pride"
 
 
-@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum", TEST_IMAGES)
+@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
 def test_checksum(
-    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, tmp_path: Path
+    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimetype: str, tmp_path: Path
 ) -> None:
     assert functions.checksum(file) == checksum
 
 
-@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum", TEST_IMAGES)
+@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
 def test_create_thumbnail(
-    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, tmp_path: Path
+    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimitype: str, tmp_path: Path
 ) -> None:
     dest_file = tmp_path / file.name
     assert not dest_file.is_file()
@@ -56,11 +56,15 @@ def test_create_thumbnail_already_exist(tmp_path: Path) -> None:
         assert dest_file.stat().st_size == 0
 
 
-@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum", TEST_IMAGES)
-def test_docolav(file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str) -> None:
+@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
+def test_docolav(
+    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimitype: str
+) -> None:
     assert functions.docolav(file) == color
 
 
-@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum", TEST_IMAGES)
-def test_get_size(file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str) -> None:
+@pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
+def test_get_size(
+    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimitype: str
+) -> None:
     assert functions.get_size(file) == size

--- a/tests/unit/test_fn_images.py
+++ b/tests/unit/test_fn_images.py
@@ -26,14 +26,28 @@ def test_any_css_class_question_gray_pride() -> None:
 
 @pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
 def test_checksum(
-    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimetype: str, tmp_path: Path
+    file: Path,
+    file_size: int,
+    size: Size,
+    thumb_size: Size,
+    color: str,
+    checksum: str,
+    mimetype: str,
+    tmp_path: Path,
 ) -> None:
     assert functions.checksum(file) == checksum
 
 
 @pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
 def test_create_thumbnail(
-    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimetype: str, tmp_path: Path
+    file: Path,
+    file_size: int,
+    size: Size,
+    thumb_size: Size,
+    color: str,
+    checksum: str,
+    mimetype: str,
+    tmp_path: Path,
 ) -> None:
     dest_file = tmp_path / file.name
     assert not dest_file.is_file()
@@ -58,13 +72,25 @@ def test_create_thumbnail_already_exist(tmp_path: Path) -> None:
 
 @pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
 def test_docolav(
-    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimitype: str
+    file: Path,
+    file_size: int,
+    size: Size,
+    thumb_size: Size,
+    color: str,
+    checksum: str,
+    mimitype: str,
 ) -> None:
     assert functions.docolav(file) == color
 
 
 @pytest.mark.parametrize("file, file_size, size, thumb_size, color, checksum, mimetype", TEST_IMAGES)
 def test_get_size(
-    file: Path, file_size: int, size: Size, thumb_size: Size, color: str, checksum: str, mimitype: str
+    file: Path,
+    file_size: int,
+    size: Size,
+    thumb_size: Size,
+    color: str,
+    checksum: str,
+    mimitype: str,
 ) -> None:
     assert functions.get_size(file) == size

--- a/tests/unit/test_fn_sync.py
+++ b/tests/unit/test_fn_sync.py
@@ -167,6 +167,7 @@ def test_sync_feed(with_timestamps: bool, tmp_path: Path) -> None:
 
     # Valid images
     for file, *_ in TEST_IMAGES[2:]:
+        print(file.name, guess_type(file)[0])
         responses.add(
             method="HEAD",
             url=f"{FEED_URL}/{file.name}",

--- a/tests/unit/test_fn_sync.py
+++ b/tests/unit/test_fn_sync.py
@@ -3,7 +3,6 @@ This is part of Shaarlimages.
 Source: https://github.com/BoboTiG/shaarlimages
 """
 
-from mimetypes import guess_type
 from pathlib import Path
 
 import pytest
@@ -114,7 +113,7 @@ def test_sync_feed(with_timestamps: bool, tmp_path: Path) -> None:
     responses.add(
         method="HEAD",
         url=f"{FEED_URL}/{TEST_IMAGES[0][0].name}",
-        content_type=guess_type(TEST_IMAGES[0][0].name)[0],
+        content_type=TEST_IMAGES[0][6],
     )
     responses.add(
         method="GET",
@@ -126,7 +125,7 @@ def test_sync_feed(with_timestamps: bool, tmp_path: Path) -> None:
     responses.add(
         method="HEAD",
         url=f"{FEED_URL}/{TEST_IMAGES[1][0].name}",
-        content_type=guess_type(TEST_IMAGES[1][0].name)[0],
+        content_type=TEST_IMAGES[1][6],
     )
     responses.add(
         method="GET",
@@ -166,12 +165,11 @@ def test_sync_feed(with_timestamps: bool, tmp_path: Path) -> None:
     )
 
     # Valid images
-    for file, *_ in TEST_IMAGES[2:]:
-        print(file.name, guess_type(file)[0])
+    for file, *_, mimetype in TEST_IMAGES[2:]:
         responses.add(
             method="HEAD",
             url=f"{FEED_URL}/{file.name}",
-            content_type=guess_type(file)[0],
+            content_type=mimetype,
         )
         responses.add(
             method="GET",

